### PR TITLE
Add Open API generator to generate angular services for API

### PR DESCRIPTION
**Pull request by pbolduc**
# Description

This PR includes the following proposed change(s):

- Contains the code to generate the typescript models and api interfaces for calling the backend.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by skiselev1 — COMMENTED** [review]
I am not sure if paths of some files are correct?

File: src/frontend/csrs-portal/src/app/api/README.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+## @
+
+### Building
+
+To install the required dependencies and to build the typescript sources run:
```
[the remark this answers is not in the document]
Comment (pbolduc) [reply src/frontend/csrs-portal/src/app/api/README.md:5]: This is a generated file.

File: src/frontend/csrs-portal/src/app/api/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+wwwroot/*.js
```
[the remark this answers is not in the document]
Comment (pbolduc) [reply src/frontend/csrs-portal/src/app/api/.gitignore:1]: This is a generated file.

File: src/frontend/csrs-portal/src/app/api/api/portalFile.service.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        }
+
+        return this.httpClient.patch<Array<object>>(`${this.configuration.basePath}/PortalFile/UpdateApplication`,
+            null,
+            {
+                params: localVarQueryParameters,
+                responseType: <any>responseType_,
+                withCredentials: this.configuration.withCredentials,
+                headers: localVarHeaders,
+                observe: observe,
```
[the remark this answers is not in the document]
Comment (pbolduc) [reply src/frontend/csrs-portal/src/app/api/api/portalFile.service.ts:296]: This is a generated file.

File: src/frontend/csrs-portal/src/app/api/api/api.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export * from './portalAccount.service';
```
[the remark this answers is not in the document]
Comment (pbolduc) [reply src/frontend/csrs-portal/src/app/api/api/api.ts:1]: the second api folder is generated automatically. We could move the files up one and adjust the commands and other support files like the generate.bat and the swagger.json file.

**Review by kkadaru — APPROVED** [review]

**Pull request merged**